x-pack/plugins/task_manager/task_store.ts

```diff
@@ -96,54 +103,86 @@ export class TaskStore {
   }
 
   /**
-   * Initializes the store, ensuring the task manager index is created and up to date.
+   * Initializes the store, ensuring the task manager index template is created
+   * and the version is up to date.
    */
   public async init() {
     if (this._isInitialized) {
       throw new Error('TaskStore has already been initialized!');
     }
 
-    const properties = {
-      type: { type: 'keyword' },
-      task: {
-        properties: {
-          taskType: { type: 'keyword' },
-          runAt: { type: 'date' },
-          interval: { type: 'text' },
-          attempts: { type: 'integer' },
-          status: { type: 'keyword' },
-          params: { type: 'text' },
-          state: { type: 'text' },
-          user: { type: 'keyword' },
-          scope: { type: 'keyword' },
-        },
-      },
-    };
+    let existingVersion = -Infinity;
+    const templateName = this.index;
+    const templateVersion = getTemplateVersion(xPackage.version);
 
     try {
-      const templateResult = await this.callCluster('indices.putTemplate', {
-        name: this.index,
-        body: {
-          index_patterns: [this.index],
-          mappings: {
-            _doc: {
-              dynamic: 'strict',
-              properties,
-            },
-          },
-          settings: {
-            number_of_shards: 1,
-            auto_expand_replicas: '0-1',
-          },
-        },
+      // check if template exists
+      const templateCheck = await this.callCluster('indices.getTemplate', {
+        name: templateName,
+        filter_path: '*.version',
       });
-      this._isInitialized = true;
-      return templateResult;
+      // extract the existing version
+      existingVersion = templateCheck[templateName] ? templateCheck[templateName].version : null;
     } catch (err) {
-      throw err;
+      if (err.statusCode !== 404) {
+        throw err; // ignore not found
+      }
+    }
+
+    if (existingVersion > templateVersion) {
+      throw new Error(
```

Comment:
Yes, it's great to think about these things now.

In the 3 preparation steps, number 2 and 3 will apply to future versions of the stack. For now, I really like the point about indexing the Kibana version in the document of the scheduled task. It will have to be the integer / template version number, so the numeric comparison can be made that will let a "new" instance claim a task scheduled by an "old" instance. I like how you wrote it in our chat conversation:  "[we] want it to so that Kibana can query for tasks with a version <= their version"

That means that the "old" instance can still start and complete Task Manager initialization, but it will be aware that there are going to be tasks that it can not claim. That definitely calls for a warning log message.